ingestion/src/metadata/ingestion/source/deltalake.py

```diff
@@ -1,127 +1,180 @@
-            service_name=config.service_name,
+            service_name=config.serviceName,
         )
         self.status = SQLSourceStatus()
+        logger.info("Establishing Sparks Session")
+        builder = (
+            pyspark.sql.SparkSession.builder.appName(self.connection_config.appName)
+            .enableHiveSupport()
+            .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+            .config(
+                "spark.sql.catalog.spark_catalog",
+                "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+            )
+        )
+        if self.connection_config.metastoreHostPort:
+            builder.config(
+                "hive.metastore.uris",
+                f"thrift://{self.connection_config.metastoreHostPort}",
+            )
+        elif self.connection_config.metastoreFilePath:
+            builder.config(
+                "spark.sql.warehouse.dir", f"{self.connection_config.metastoreFilePath}"
+            )
+        self.spark = configure_spark_with_delta_pip(builder).getOrCreate()
+        self.table_type_map = {
+            TableType.External.value.lower(): TableType.External.value,
+            TableType.View.value.lower(): TableType.View.value,
+            TableType.SecureView.value.lower(): TableType.SecureView.value,
+            TableType.Iceberg.value.lower(): TableType.Iceberg.value,
+        }
+        self.array_datatype_replace_map = {"(": "<", ")": ">", "=": ":", "<>": ""}
+        self.ARRAY_CHILD_START_INDEX = 6
+        self.ARRAY_CHILD_END_INDEX = -1
 
     @classmethod
-    def create(cls, config_dict: dict, metadata_config: OpenMetadataServerConfig):
-        config = DeltaLakeSourceConfig.parse_obj(config_dict)
+    def create(cls, config_dict, metadata_config: OpenMetadataServerConfig):
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: DeltaLakeConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, DeltaLakeConnection):
+            raise InvalidSourceException(
+                f"Expected DeltaLakeConnection, but got {connection}"
+            )
         return cls(config, metadata_config)
 
     def next_record(self) -> Iterable[OMetaDatabaseAndTable]:
         schemas = self.spark.catalog.listDatabases()
         for schema in schemas:
-            if not self.config.schema_filter_pattern.included(schema):
-                self.status.filter(schema, "Schema pattern not allowed")
+            if filter_by_schema(
+                self.config.sourceConfig.config.schemaFilterPattern, schema.name
+            ):
+                self.status.filter(schema.name, "Schema pattern not allowed")
                 continue
-            yield from self.fetch_tables(schema)
+            yield from self.fetch_tables(schema.name)
+
+    def get_status(self):
+        return self.status
+
+    def prepare(self):
+        return super().prepare()
```

Comment:
Is this needed?

Reply:
@pmbrull, we are inheriting the abstract class `Source` in deltalake, so If I don't add this definition here I get the following Error

```
(venv) mayursingal@Mayurs-MacBook-Pro ingestion % metadata ingest -c examples/workflows/deltalake.json 
Traceback (most recent call last):
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/bin/metadata", line 8, in <module>
    sys.exit(metadata())
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/metadata/cmd.py", line 90, in ingest
    workflow = Workflow.create(workflow_config)
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/metadata/ingestion/api/workflow.py", line 136, in create
    return cls(config)
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/metadata/ingestion/api/workflow.py", line 56, in __init__
    self.source: Source = source_class.create(
  File "/Users/mayursingal/Desktop/ulixius1/OpenMetadata/venv/lib/python3.8/site-packages/metadata/ingestion/source/deltalake.py", line 95, in create
    return cls(config, metadata_config)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/typing.py", line 873, in __new__
    obj = super().__new__(cls)
TypeError: Can't instantiate abstract class DeltalakeSource with abstract methods prepare
```

Reply:
ah I see I see, got mixed up with sql_source instead of source. Then followup question. Any reason to extend source directly instead of SQLSource? 😅  We can merge this in, but let's keep in mind if this needs a restructure at some point